### PR TITLE
Feature/xpack info usage

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

File: src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		public XPackBuildInformation Build { get; internal set; }
+		public MinimalLicenseInformation License { get; internal set; }
+		public XPackFeatures Features { get; internal set; }
+		public string Tagline { get; internal set; }
+	}
+
+	public class XPackBuildInformation
+	{
+		[JsonProperty("date")]
+		public DateTime Date { get; internal set; }
```
Comment (russcam) [inline src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs]: thoughts on using `DateTimeOffset`?

File: src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		public LicenseType Type { get; internal set; }
+
+		[JsonProperty("mode")]
+		public LicenseType Mode { get; internal set; }
+
+		[JsonProperty("status")]
+		public LicenseStatus Status { get; internal set; }
+
+		[JsonProperty("expiry_date_in_millis")]
+		public long ExpiryDateInMilliseconds { get; set; }
```
Comment (russcam) [inline src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs]: Should we expose a `DateTime`/`DateTimeOffset` property for this?

File: src/Nest/_Generated/_Descriptors.generated.cs [hunk 3]
```diff
@@ -6561,56 +6561,50 @@ public partial class GraphExploreDescriptor<T>  : RequestDescriptorBase<GraphExp
 	}
 	
 	///<summary>descriptor for XpackInfo <pre>Retrieve information about xpack, including build number/timestamp and license status</pre></summary>
-	public partial class XpackInfoDescriptor  : RequestDescriptorBase<XpackInfoDescriptor,XpackInfoRequestParameters, IXpackInfoRequest>, IXpackInfoRequest
+	public partial class XPackInfoDescriptor  : RequestDescriptorBase<XPackInfoDescriptor,XPackInfoRequestParameters, IXPackInfoRequest>, IXPackInfoRequest
```
Comment (russcam) [inline src/Nest/_Generated/_Descriptors.generated.cs]: These change the casing of `Xpack` -> `XPack`. Should we keep the casing as is here for bwc and update in `master`?

Comment (Mpdreamz) [inline src/Nest/_Generated/_Descriptors.generated.cs]: They change the casing against the generated code in `5.4`. `5.x` and `master` do not have this yet so we're good 👍

File: src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs (hunk 1 above)
Comment (Mpdreamz) [inline src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs]: ++

File: src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs (hunk 2 above)
Comment (Mpdreamz) [inline src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs]: I'm going to leave this as, might be useful for `elasticsearch` itself to start returning `date`.

File: src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs (hunk 1 above)
Comment (Mpdreamz) [inline src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs]: Added 👍

**Comment by Mpdreamz** [comment]
ported to `master`

**Pull request merged**